modules/page-memory/src/main/java/org/apache/ignite/internal/pagememory/persistence/PersistentPageMemory.java

```diff
@@ -587,10 +587,9 @@ public long allocatePageNoReuse(int grpId, int partId, byte flags) throws Ignite
             throw e;
         } finally {
             seg.writeLock().unlock();
-        }
 
-        // Finish replacement only when an exception wasn't thrown otherwise it possible to corrupt B+Tree.
```

Comment:
I don't think so, because now even if an exception is thrown, we will end up replacing the page to finish the checkpoint if possible.
What do you think?

Reply:
Ok, let's remove it, I don't mind